xray/core/dataset.py

```diff
@@ -804,17 +804,26 @@ def reset_coords(self, names=None, drop=False, inplace=False):
                 del obj._variables[name]
         return obj
 
-    def dump_to_store(self, store, encoder=None, sync=True):
+    def dump_to_store(self, store, encoder=None, sync=True, encoding={}):
         """Store dataset contents to a backends.*DataStore object."""
         variables, attrs = conventions.encode_dataset_coordinates(self)
+
+        check_encoding = set()
+        for k, enc in encoding.items():
```

Comment:
I tend to prefer using `.items()` instead of `iteritems()` because it's less verbose and just does the right thing on Python 3. I think `iteritems()` is only worth using in cases where it actually matters for performance.

To be honest, though, I haven't really applied these rules consistently.